cupy/cuda/compiler.py

```diff
@@ -276,18 +276,26 @@ def get_cache_dir():
 _empty_file_preprocess_cache = {}
 
 
-def compile_with_cache(source, options=(), arch=None, cache_dir=None,
-                       extra_source=None, backend='nvrtc'):
+def compile_with_cache(
+        source, options=(), arch=None, cache_dir=None, extra_source=None,
+        backend='nvrtc', *, enable_cooperative_groups=False):
+
+    if enable_cooperative_groups and backend != 'nvcc':
+        raise ValueError(
+            'Cooperative groups is supported only in NVCC backend.')
+
     if runtime.is_hip:
-        return _compile_with_cache_hipcc(source, options, arch, cache_dir,
-                                         extra_source)
+        return _compile_with_cache_hipcc(
+            source, options, arch, cache_dir, extra_source)
```

Comment:
How about raising an error in this case?

Reply:
Fixed.